birdhouse/components/README.rst

```diff
@@ -135,6 +135,131 @@ Features missing in old install scripts or how the new mechanism improves on the
+
+  * This could fail if some changes in the PR, are incompatible with autodeploy. For example: ``./pavics-compose.sh`` calls some binaries that do not exist in the autodeploy docker image.
+
+* Can autodeploy be triggered again successfully, after the PR is live
+
+  * This could fail if the PR renamed some files and forgot to add the old file names to a ``.gitignore`` file.  Then old file names will appear as new uncommitted files and autodeploy will halt because it expects a clean working directory.
+
+Here is a sample setup to test autodeploy:
+
+* Have 2 checkout directories.  One is for starting the stack using ``./pavics-compose.sh``, the other one is to push new bogus changes to trigger the autodeploy mechanism.
+
+.. code-block:: shell
+
+  # this one for running pavics-compose.sh
+  git clone git@github.com:bird-house/birdhouse-deploy.git birdhouse-deploy
+
+  # this one for triggering autodeploy
+  git clone git@github.com:bird-house/birdhouse-deploy.git birdhouse-deploy-trigger
+
+* Set ``AUTODEPLOY_PLATFORM_FREQUENCY`` in ``env.local`` to a very frequent value so you do not have to wait too long for autodeploy to trigger.
+
+.. code-block:: shell
+
+  # go to the main checkout
+  cd birdhouse-deploy/birdhouse
+
+  # ensure the scheduler component is enabled, otherwise autodeploy will not work
+  echo 'export EXTRA_CONF_DIRS="$EXTRA_CONF_DIRS ./components/scheduler" >> env.local
+
+  # set AUTODEPLOY_PLATFORM_FREQUENCY
+  echo 'export AUTODEPLOY_PLATFORM_FREQUENCY="@every 5m"' >> env.local
+
+  # recreate scheduler container for new AUTODEPLOY_PLATFORM_FREQUENCY to be effective
+  ./pavics-compose.sh stop scheduler && ./pavics-compose.sh rm -vf scheduler && ./pavics-compose.sh up -d
+
+* Create a ``test`` branch so you can add bogus commits without affecting your real PR.  Set up your main checkout (birdhouse-deploy) to track that test branch so it will detect new changes on the test branch and trigger the autodeploy.
+
+.. code-block:: shell
+
+  # go to the main checkout
+  cd birdhouse-deploy/birdhouse
+
+  # initially create the test branch from master
+  git checkout master
+  git pull
+  git checkout -b test
+  git push -u test
+
+  # ensure your runnings code is at "master" and is working correctly
+  # if you do not have a working baseline, you will not know if the breakage is due to autodeploy or your code
+  ./pavics-compose.sh up -d
+
+* Test scenario 1, from ``master`` to your PR
+
+.. code-block:: shell
+
+  # go to the other checkout to trigger autodeploy
+  cd birdhouse-deploy-trigger/birdhouse
+
+  # set branch test to the same commit as your PR, this will trigger autodeploy from master to your PR
+  git pull
+  git checkout test
+  git reset --hard YOUR_PR_BRANCH
+  git push
+
+  # now that the remote "test" branch differs from the local "test" branch in the birdhouse-deploy repo,
+  # the autodeploy mechanism will detect that the remote branch has changed and attempt to update the local branch
```

Comment:
Is there a specific log message somewhere that can help the user identify that autodeploy was started/running/applied?
If so, it could be useful to provide it here to guide the user about what to look for.

Reply:
In /var/log/PAVICS/autodeploy.log, each autodeploy trigger will start with
```
==========
triggerdeploy START_TIME=2023-06-15T05:07:01+0000
```
and ends with
```
triggerdeploy finished START_TIME=2023-06-15T05:07:01+0000
triggerdeploy finished   END_TIME=2023-06-15T05:07:06+0000
```